src/dirtyfields/dirtyfields.py

```diff
@@ -83,6 +88,9 @@ def get_dirty_fields(self, check_relationship=False, check_m2m=None, verbose=Fal
             initial_dict = self._as_dict(check_relationship, include_primary_key=False)
             return initial_dict
 
+        if check_m2m is not None and not self.ENABLE_M2M_CHECK:
+            raise Exception("You can't check m2m fields if ENABLE_M2M_CHECK is set to False")
```

Comment:
A `ValueError` would be appropriate here, since the user has passed in invalid argument.

> exception ValueError
Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.

https://docs.python.org/3/library/exceptions.html#ValueError